index.js

```diff
@@ -80,37 +81,66 @@ async function annotatePackage(packageName, filePath, lineNumber) {
     }
 }
 
-async function run() {
-    const filePath = 'requirements.txt';
-    const ecosystem = core.getInput('package-ecosystem', { required: true });
+async function processLines(filePath, lines) {
+    for (let index = 1; index <= lines.length; index++) {
+        const line = lines[index - 1]; // Keep lines including empty ones
 
-    if (ecosystem !== 'pip') {
-        core.setFailed(`Unsupported package ecosystem: ${ecosystem}`);
-        return;
+        if (!line.trim()) {
+            // Skip processing empty lines, but continue incrementing line count
+            continue;
+        }
+
+        const packageName = processPackageLine(line);
+        if (packageName) {
+            if (!isValidPackageName(packageName)) {
+                core.error(`Invalid package name: ${packageName}`, {
+                    file: filePath,
+                    startLine: index,
+                    endLine: index
+                });
+            } else {
+                await annotatePackage(packageName, filePath, index);
+            }
+        }
     }
+}
 
+async function processPipRequirements(filePath) {
     try {
-        const packages = (await fs.readFile(filePath, 'utf-8')).split('\n').filter(pkg => pkg);
-
-        packages.forEach((packageLine, index) => {
-            const packageName = processPackageLine(packageLine);
-            const lineNumber = index + 1;
-
-            if (packageName) {
-                if (!isValidPackageName(packageName)) {
-                    core.error(`Invalid package name: ${packageName}`, {
-                        file: filePath,
-                        startLine: lineNumber,
-                        endLine: lineNumber
-                    });
-                    return;
-                }
-                annotatePackage(packageName, filePath, lineNumber);
-            }
-        });
+        const lines = (await fs.readFile(filePath, 'utf-8')).split('\n');
+        await processLines(filePath, lines);
     } catch (error) {
         core.setFailed(`Failed to read ${filePath}: ${error.message}`);
     }
 }
 
+async function processCondaEnvironment(filePath) {
+    try {
+        const fileContent = await fs.readFile(filePath, 'utf-8');
+        const environment = yaml.load(fileContent);
+
+        if (environment?.dependencies) {
+            const lines = fileContent.split('\n');
+            await processLines(filePath, lines);
```

Comment:
what is this doing? this does not look robust. please implement this so it is robust to errors and/or document where the shortcommings are

eg 
```
for package in environment.dependencies:
   const lineNo = getLineNo(package)
```  
what if this is in yaml flow notation also won;t your soltion tread conda channels as deps?

```
channels:
  - conda-forge
```